src/generic_core/remote-instance.spec.ts

```diff
@@ -111,8 +111,9 @@ describe('remote_instance.RemoteInstance', () => {
     var INSTANCE_ID = 'instance1';
 
     beforeEach((done) => {
-      globals.storage = new local_storage.Storage();
-      globals.storage.reset().then(() => {
+      let testStorage = new local_storage.Storage();
+      globals.setGlobalStorageForTest(testStorage);
```

Comment:
Notice how I had to manually edit this. The es6 imports are constant, and exposed that we were modifying a global here. I had to add a setter as a workaround. Same for the nat code.